### PR TITLE
evaluate-using-a-service-mesh.md

### PR DESCRIPTION
**Pull request by yelghali**
@dubrie

### CODE REVIEW

File: docs/catalog/cloud/evaluate-using-a-service-mesh.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+version: 1.0
+submitted_by: yelghali
+published_date: tbd
+category: cloud
+tags: 
+- security
```
Comment (dubrie) [inline docs/catalog/cloud/evaluate-using-a-service-mesh.md:7]: please include `cloud` in the tags list as well.

File: docs/catalog/cloud/evaluate-using-a-service-mesh.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+- network
+- kubernetes
+- role:cloud-engineer
+- size:medium
+---
+
+# Evaluate using a service mesh
```
Comment (dubrie) [inline docs/catalog/cloud/evaluate-using-a-service-mesh.md:14]: Pattern titles should be more action-oriented on a resource and any "evaluations" should be part of the assumptions or considerations sections below.

File: docs/catalog/cloud/evaluate-using-a-service-mesh.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+## Description
+A service mesh deploys additional containers for communication, typically in a sidecar pattern, to provide more operational capabilities leading to an increase in CPU usage and network traffic. 
+
+Nevertheless, it allows you to decouple your application from these capabilities as it moves them out from the application layer, and down to the infrastructure layer.
+
+## Solution
+Carefully consider the increase in CPU usage and network traffic generated by service mesh communication components before making the decision to use one.
```
Comment (dubrie) [inline docs/catalog/cloud/evaluate-using-a-service-mesh.md:22]: What are the considerations a cloud engineer should be accounting for? How should they be weighing the service-mesh decision against functional requirements and emissions?

File: docs/catalog/cloud/evaluate-using-a-service-mesh.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+Regarding the SCI equation, not using a service mesh will impact:
+
+- `E`: By reducing CPU utilization and network traffic, we reduce the amount of energy required.
+- `M`: reducing CPU utilization and network data reduces the amount of resources required, which decreases the embodied carbon emissions.
+
+## Assumptions
+The application does not have compliance requirements for using a service mesh. 
+
+## Considerations
```
Comment (dubrie) [inline docs/catalog/cloud/evaluate-using-a-service-mesh.md:37]: I think some of this should be brought into the solution section. It's not clear based on the current content what an engineer should do in this scenario or what they should be considering. This section seems to highlight some of the actions or steps they can take into account.

**Review by dubrie — CHANGES_REQUESTED** [review]
@yelghali Some updates requested from the initial review

**Comment by yelghali** [comment]
@dubrie thank you, done.

File: docs/catalog/cloud/evaluate-using-a-service-mesh.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+# Use a service mesh only if needed
+
+## Description
+A service mesh deploys additional containers for communication, typically in a sidecar pattern, to provide more operational capabilities leading to an increase in CPU usage and network traffic. 
+
+Nevertheless, it allows you to decouple your application from these capabilities as it moves them out from the application layer, and down to the infrastructure layer.
+
+## Solution
+- Use a service mesh if your application lacks features that can be added by one, such as distributed tracing or traffic encryption.
+- A service mesh can typically be enabled only for some resources, which allows you to use it for required applications only, and not the entire platform.
```
Comment (markus-ntt-seidl) [inline docs/catalog/cloud/evaluate-using-a-service-mesh.md:24]: Some things can also be done inside the application itself, without using a service mesh. For example, other service DNS names could be injected via configuration
For example:
* a ConfigMap -> Environment Variable in Kubernetes or
* A custom configured DNS Server that answers with the correct IPs (DNS servers are mostly needed in private/corporate environments)

File: docs/catalog/cloud/evaluate-using-a-service-mesh.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+Regarding the SCI equation, not using a service mesh will impact:
+
+- `E`: By reducing CPU usage and network traffic, we reduce the amount of energy required.
+- `M`: reducing CPU usage and network data reduces the amount of resources required, which decreases the embodied carbon emissions.
+
+## Assumptions
+The application does not have compliance requirements for using a service mesh. 
+
+## Considerations
+Should you use a service mesh ? The answer depends on your requirements.
```
Comment (markus-ntt-seidl) [inline docs/catalog/cloud/evaluate-using-a-service-mesh.md:40]: I agree that this needs either put into the solution section, or be rephrased into some more fitting.
For example:
- Consider alternative methods outside of an service mesh to fulfill application requirements, for example custom DNS servers or configuration that is injected into the application

File: docs/catalog/cloud/evaluate-using-a-service-mesh.md (hunk 5 above)
Comment (franziska-warncke) [reply docs/catalog/cloud/evaluate-using-a-service-mesh.md:24]: How would you add this here in this solution section?

**Comment by franziska-warncke** [comment]
@markus-gsf-seidl Please have a look again.

File: docs/catalog/cloud/evaluate-using-a-service-mesh.md (hunk 5 above)
Comment (markus-gsf-seidl) [reply docs/catalog/cloud/evaluate-using-a-service-mesh.md:24]: Putting this under considerations is fine for me.

**Review by markus-gsf-seidl — APPROVED** [review]
LGTM

**Pull request merged**